specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json

```diff
@@ -0,0 +1,2964 @@
+      "x-ms-client-flatten": true,
+      "type": "object",
+      "properties": {
+        "principalId": {
+          "type": "string",
+          "description": "ObjectId from the KeyVault",
+          "readOnly": true
+        },
+        "tenantId": {
+          "type": "string",
+          "description": "TenantId from the KeyVault",
+          "readOnly": true
+        },
+        "type": {
+          "type": "string",
+          "description": "Type of managed service identity.",
+          "enum": [
+            "UserAssigned"
+          ],
+          "x-ms-enum": {
+            "name": "ManagedServiceIdentityType",
+            "modelAsString": false
+          }
+        },
+        "userAssignedIdentities": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "object",
+            "description": "",
+            "items": {
+              "$ref": "#/definitions/UserAssignedIdentity"
+            }
+          },
+          "description": "Metadata of user assigned identity."
+        }
+      },
+      "description": "Properties to configure Identity for Bring your Own Keys"
+    },
+    "UserAssignedIdentity": {
+      "type": "object",
+      "properties": {
+        "principalId": {
+          "description": "Principal Id of user assigned identity",
+          "type": "string",
+          "x-ms-client-name": "PrincipalId",
+          "readOnly": true
+        },
+        "clientId": {
+          "description": "Client Id of user assigned identity",
+          "type": "string",
+          "x-ms-client-name": "ClientId",
+          "readOnly": true
+        }
+      },
+      "description": "Metadata of user assigned identity."
+    },
+    "CutoverMigrationRequest": {
+      "type": "object",
+      "description": "Cutover migration request",
+      "properties": {
+        "dnsChangeMode": {
+          "type": "string",
+          "enum": [
+            "Update",
+            "Revert",
+            "None"
+          ],
+          "x-ms-enum": {
+            "name": "DnsChangeMode",
+            "modelAsString": true
+          },
+          "description": "The DNS Mode"
+        }
+      }
+    },
+    "MigrationAssessmentRequest": {
+      "type": "object",
+      "properties": {
+        "sourceServerResourceId": {
+          "type": "string",
+          "description": "Source Server Resource Id to verify and perform migration assessment on."
+        }
+      },
+      "description": "Request from client to perform migration assessment"
+    },
+    "MigrationAssessmentResource": {
+      "type": "object",
+      "properties": {
+        "sourceServerResourceId": {
+          "type": "string",
+          "description": "Source Server Resource Id to verify and perform migration assessment on."
+        },
+        "targetPlatformProviderId": {
+          "type": "string",
+          "description": "Target platform(Flexible Server) provider ID"
+        },
+        "targetServerDetails": {
```

Comment:
>targetServerDetails [](http://example.com/codeflow?start=9&length=19)

I'm finding source and target to be a getting a little non-intuitive for these properties in the face of reports containing replica reports. So a report resource has a source server, a target server and an array of source replicas. Each source replica also has a source server, target server and an array of source replicas. What's in the replica array of a replica report?

Assuming I understand your API correctly, I think it would be a little clearer if you have a single machine assessment report definition and an outer assessment report definition that uses the other definition to lay out a primary and array of replicas.

Something more like:

machineAssessment: {
  serverResourceId,
  migrationInformation
}

assessmentReport: {
  primary: machineAssessment
  replicas: [
    machineAssessment,
    machineAssessment,
   ...
  ]
}

Reply:
For your first question ie;
1.Each source replica also has a source server(represents the replica server), target server(target server for replica) and an array of source replicas. What's in the replica array of a replica report?
And: The replica array will remain empty for the source replica part.

2.My thoughts based on the new model you suggested
Here the primary can be the replica as well. By this what I mean is, the customer can call migration assessment directly on the replica as well.
In this case the model we defined makes more sense. WDYT?

Reply:
Taking all comments into perspective, I think it works well if you change the name "primary" to "server" in the above structure. This also avoids creating a recursive definition. Either approach returns an empty replica array when the request is performed on a replica.

Reply:
I still find the source and target terms pretty confusing. My suggestion is to use the structure rather than source/target to clarify the meaning, but I won't block on this.